packages/medusa/src/strategies/__tests__/tax-calculation.js

```diff
@@ -1,124 +1,177 @@
-
-    /*
-     * Subtotal = 2 * 100 = 200
-     * Taxable amount = 200 - 10 = 180
-     * Taxline 1 = 180 * 0.0825 = 15
-     * Taxline 2 = 180 * 0.125 = 23
-     * Total tax = 38
-     */
-    expected: 40,
-    items: [
-      {
-        id: "item_1",
-        unit_price: 100,
-        quantity: 2,
-      },
-    ],
-    taxLines: [
-      {
-        item_id: "item_1",
-        name: "Name 1",
-        rate: 8.25,
-      },
-      {
-        item_id: "item_1",
-        name: "Name 2",
-        rate: 12.5,
-      },
-    ],
-    context: {
-      shipping_address: null,
-      customer: {
-        email: "test@testson.com",
-      },
-      region: {
-        gift_cards_taxable: true,
-      },
-      shipping_methods: [],
-      allocation_map: {
-        item_1: {
-          discount: {
-            amount: 10,
-            unit_amount: 5,
-          },
-          gift_card: {
-            amount: 10,
-            unit_amount: 5,
+  ],
+  [
+    "calculates correctly with gift card",
+    {
+      /*
+       * Subtotal = 2 * 100 = 200
+       * Taxable amount = 200 - 10 = 180
+       * Taxline 1 = 180 * 0.0825 = 15
+       * Taxline 2 = 180 * 0.125 = 23
+       * Total tax = 38
+       */
+      expected: 40,
+      items: [
+        {
+          id: "item_1",
+          unit_price: 100,
+          quantity: 2,
+        },
+      ],
+      taxLines: [
+        {
+          item_id: "item_1",
+          name: "Name 1",
+          rate: 8.25,
+        },
+        {
+          item_id: "item_1",
+          name: "Name 2",
+          rate: 12.5,
+        },
+      ],
+      context: {
+        shipping_address: null,
+        customer: {
+          email: "test@testson.com",
+        },
+        region: {
+          gift_cards_taxable: true,
+        },
+        shipping_methods: [],
+        allocation_map: {
+          item_1: {
+            discount: {
+              amount: 10,
+              unit_amount: 5,
+            },
+            gift_card: {
+              amount: 10,
+              unit_amount: 5,
+            },
           },
         },
       },
     },
-  },
+  ],
 ]
 
+const taxInclusiveShippingTests = [
+  [
+    "calculates correctly with tax inclusive shipping",
+    {
+      expected: 25,
+      items: [],
+      taxLines: [
+        {
+          shipping_method_id: "shipping_method_1",
+          name: "Name 1",
+          rate: 25,
+        },
+        {
+          shipping_method_id: "shipping_method_2",
+          name: "Name 2",
+          rate: 10,
+        },
+      ],
+      context: {
+        shipping_methods: [
+          { id: "shipping_method_1", price: 100, includes_tax: true },
+          { id: "shipping_method_2", price: 50, includes_tax: false },
+        ],
+      },
+    },
+  ],
+]
 describe("TaxCalculationStrategy", () => {
-  describe("calculate", () => {
-    const calcStrat = new TaxCalculationStrategy()
+  ;[true, false].forEach((taxInclusivePricingEnabled) => {
```

Comment:
comment: could I let you look at this https://github.com/medusajs/medusa/pull/2123/files#diff-34799757e3c3872bad5aaff5646e430301f287cfe7e87b98ead5d5f4cc5da4aa and let me know what you think. basically, the flags can be specified through the configuration object of each tests

Reply:
yeah I considered it, I opted for running all the tests with flags enabled and disabled for good measure 😄

Reply:
I think they are different approaches and we need to find a way to unite them I guess 😄

Reply:
Ahah indeed, my thinking was that it would support different flags settings ☺️

Reply:
I have undone my changes to tax-calculation strategy in favor of @carlos-r-l-rodrigues pr 😄